plugins/woocommerce/includes/gateways/cheque/class-wc-gateway-cheque.php

```diff
@@ -113,10 +115,12 @@ public function email_instructions( $order, $sent_to_admin, $plain_text = false
 		 * Filter the email instructions order status.
 		 *
 		 * @since 7.4
-		 * @param string $terms The order status.
-		 * @param object $order The order object.
+		 *
+		 * @param string $status The default status.
+		 * @param object $order  The order object.
 		 */
-		if ( $this->instructions && ! $sent_to_admin && 'cheque' === $order->get_payment_method() && $order->has_status( apply_filters( 'woocommerce_cheque_email_instructions_order_status', 'on-hold', $order ) ) ) {
+		$instructions_order_status = apply_filters( 'woocommerce_cheque_email_instructions_order_status', OrderStatus::ON_HOLD, $order );
```

Comment:
This way, the filter invocation is a bit preliminary, so it might be more invocations than necessary.

I'm suggesting refining approach as follows to address that:
```
if ( $this->instructions && ! $sent_to_admin && 'cheque' === $order->get_payment_method() ) {
   // doc-block here
   $instructions_order_status = apply_filters( 'woocommerce_cheque_email_instructions_order_status', OrderStatus::ON_HOLD, $order );
    if ( $order->has_status( $instructions_order_status )  ) {
        echo wp_kses_post( wpautop( wptexturize( $this->instructions ) ) . PHP_EOL );
    }
}
```

Reply:
It might be more similar patterns in the PR, but let's align on this one first.

Reply:
Makes sense @kalessil . I have addressed it in https://github.com/woocommerce/woocommerce/pull/51922/commits/af592ec50cc9e35e0a3dc9d6e884b7cad922e40f